src/routers/UpdateAvailabilitiesRouter.ts

```diff
@@ -0,0 +1,77 @@
+import { Request } from 'express';
+import constants from '../common/constants';
+import Availability from '../entities/Availability';
+import AvailabilityRepo from '../repos/AvailabilityRepo';
+import UserRepo from '../repos/UserRepo';
+import AuthenticatedAppplicationRouter from '../utils/AuthenticatedApplicationRouter';
+
+class UpdateAvailabilitiesRouter extends AuthenticatedAppplicationRouter<void> {
+  constructor() {
+    super('POST');
+  }
+
+  getPath(): string {
+    return '/availabilities/';
+  }
+
+  async content(req: Request): Promise<void> {
+    const { user, body } = req;
+
+    const validFields = ['schedule'];
+
+    Object.keys(body).forEach((key) => {
+      if (!validFields.includes(key)) {
+        throw Error(`Invalid field '${key}' identified in request body.`);
```

Comment:
If all of the necessary fields are present, is it really worth failing over an extra one? I don't think I've ever seen an api do this before.

Reply:
I was following the format of the other update routes. I was thinking it's necessary to have the data look exactly as it does in the User entity but it looks like db().merge is smarter than that. I'll remove these strict checks on this route.

Reply:
^ specifically, I got that from updateDemographics, which sets single fields and so we would want to make sure we don't touch interests, clubs, or availabilities since they require additional processing and have their own routes, so this doesn't seem to be an issue in the other routers.

Reply:
Hmm interesting. I would have approached that by copying the information I need specifically into a new object to pass to other functions instead of using the body directly, but I guess failing if the body doesn't match works as well.